### ISSUE TITLE
S108 does not know about the effects of `using` statements

### ISSUE THREAD

**Issue by jhinder** [SonarSource/sonar-dotnet#7016]
### Description

S108 ("Nested blocks of code should not be left empty") should not trigger if the block belongs to a `using` statement.

There was no clear consensus among our development team. It can be seen as a false-positive because the statement actually does something even with an empty block; on the other hand an empty block of code can always be seen as suspicious and should at least be documented/commented so that it's clear that this block is empty on purpose.
For this reason we'd like to get feedback from you on this matter: should there be an exception for statements with clearly defined side effects, or should the general rule stand and developers must suppress them in some way to acknowledge that they're aware of this smell?

### Repro steps

```csharp
using System.Diagnostics;
class C
{
    void M()
    {
        using (Process.Start(@"C:\Windows\notepad.exe"))
        {
        }
    }
}
```

The method call returns an IDisposable instance, but we're not interested in the instance and just want to dispose of it right away. Using `using` syntax is easier than manually writing the equivalent try/finally statement.

### Expected behavior

No diagnostic; the block is empty, but the statement as a whole still performs the usual work of a `using` statement.

### Actual behavior

Diagnostic S108 is emitted for the block syntax node.

### Known workarounds

* Add a comment (documented exception for the rule).
* Rewrite the statement to use a `using` declaration (`using var _ = …;`).

### Related information

* C# plugin 8.55.0.65544
* Visual Studio 17.5.3
* .NET 7.0.4, dotnet sdk 7.0.202
* Windows 11

**Comment by mary-georgiou-sonarsource** [comment]
Thank you for opening this issue, @jhinder . Your feedback helps us improve our products. We will look at it as soon as possible. After we confirm the issue, we will add labels and put it in our backlog

**Comment by zsolt-kolbay-sonarsource** [comment]
Hi @jhinder. Other than the workarounds you mentioned, you can also use the `Dispose()` method directly. That way you can avoid creating a new variable or an empty code block:
`Process.Start(@"C:\Windows\notepad.exe").Dispose();`
or if it can return null
`Process.Start(@"C:\Windows\notepad.exe")?.Dispose();`
